crates/services/txpool_v2/src/tests/tests_pending_pool.rs

```diff
@@ -0,0 +1,128 @@
+use fuel_core_services::Service;
+use fuel_core_types::{
+    fuel_tx::{
+        UniqueIdentifier,
+        UtxoId,
+    },
+    services::txpool::TransactionStatus,
+};
+use futures::StreamExt;
+
+use crate::{
+    tests::universe::TestPoolUniverse,
+    TxStatusMessage,
+};
+
+#[tokio::test]
+async fn test_tx_keep_missing_input() {
+    let mut universe = TestPoolUniverse::default();
+    universe.config.utxo_validation = true;
+
+    let (output, unset_input) = universe.create_output_and_input();
+    let tx1 = universe.build_script_transaction(None, Some(vec![output]), 10);
+    let tx_id1 = tx1.id(&Default::default());
+    let input = unset_input.into_input(UtxoId::new(tx_id1, 0));
+    let tx2 = universe.build_script_transaction(Some(vec![input]), None, 20);
+
+    let service = universe.build_service(None, None);
+    service.start_and_await().await.unwrap();
+
+    // Given
+    service.shared.try_insert(vec![tx2.clone()]).unwrap();
+
+    // When
+    tokio::time::timeout(
+        tokio::time::Duration::from_secs(1),
+        universe.waiting_txs_insertion(
+            service.shared.new_tx_notification_subscribe(),
+            vec![tx2.id(&Default::default())],
+        ),
+    )
+    .await
+    .unwrap_err();
+    service.shared.try_insert(vec![tx1.clone()]).unwrap();
+
+    universe
+        .waiting_txs_insertion(
+            service.shared.new_tx_notification_subscribe(),
+            vec![tx1.id(&Default::default()), tx2.id(&Default::default())],
+        )
+        .await;
+
+    // Then
+    service.stop_and_await().await.unwrap();
+}
+
+#[tokio::test]
+async fn test_tx_kept_expired() {
+    let mut universe = TestPoolUniverse::default();
+    universe.config.utxo_validation = true;
+    let timeout = tokio::time::Duration::from_millis(100);
+    universe.config.pending_pool_tx_ttl = timeout;
+
+    let (_, unset_input) = universe.create_output_and_input();
+    let tx1 = universe.build_script_transaction(None, None, 10);
+    let tx_id1 = tx1.id(&Default::default());
+    let input = unset_input.into_input(UtxoId::new(tx_id1, 0));
+    let tx2 = universe.build_script_transaction(Some(vec![input]), None, 20);
+    let tx_id2 = tx2.id(&Default::default());
+
+    let service = universe.build_service(None, None);
+    service.start_and_await().await.unwrap();
+
+    // Given
+    service.shared.try_insert(vec![tx2.clone()]).unwrap();
+    let mut subscriber_status = service.shared.tx_update_subscribe(tx_id2).unwrap();
+
+    // When
+    tokio::time::timeout(
+        timeout,
+        universe.waiting_txs_insertion(
+            service.shared.new_tx_notification_subscribe(),
+            vec![tx2.id(&Default::default())],
+        ),
+    )
+    .await
+    .unwrap_err();
+    service.shared.try_insert(vec![tx1.clone()]).unwrap();
```

Comment:
I think we want to sleep here for 1 second to be sure that it will be not included

Reply:
If you meant before the insert of tx1 it's already done by the timeout. If you mean after, we already sure that it has been squeezed out because we are waiting for this status in the subscriber.